Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp

```diff
@@ -1003,7 +1030,7 @@ RefPtr<WebCore::ShapeDetection::FaceDetector> WebChromeClient::createFaceDetecto
 RefPtr<WebCore::ShapeDetection::TextDetector> WebChromeClient::createTextDetector() const
 {
 #if ENABLE(GPU_PROCESS)
-    auto& remoteRenderingBackendProxy = m_page.ensureRemoteRenderingBackendProxy();
+    auto& remoteRenderingBackendProxy = protectedPage()->ensureRemoteRenderingBackendProxy();
```

Comment:
This doesn't seem right. Why is it safe to hold the reference from this function without protecting the page in a local variable?